pkg/volume/empty_dir/empty_dir.go

```diff
@@ -254,8 +255,36 @@ func (ed *emptyDir) setupTmpfs(dir string) error {
 		return nil
 	}
 
-	glog.V(3).Infof("pod %v: mounting tmpfs for volume %v", ed.pod.UID, ed.volName)
-	return ed.mounter.Mount("tmpfs", dir, "tmpfs", nil /* options */)
+	memoryLimits, err := ed.getMemoryLimits()
+	if err != nil {
+		return err
+	}
+	var options []string
+	if memoryLimits > 0 {
+		options = []string{fmt.Sprintf("size=%d", memoryLimits)}
+	}
+	glog.V(3).Infof("pod %v: mounting tmpfs for volume %v with options %v", ed.pod.UID, ed.volName, options)
+	return ed.mounter.Mount("tmpfs", dir, "tmpfs", options)
+}
+
+func (ed *emptyDir) getMemoryLimits() (int64, error) {
+	if string(qos.GetPodQOS(ed.pod)) == "Guaranteed" {
+		_, limits, err := v1.PodRequestsAndLimits(ed.pod)
+		if err != nil {
+			return 0, err
+		}
+		val := limits[v1.ResourceMemory]
+		return (&val).Value(), nil
+	}
+	// For all other QOS class use allocatable as limit.
+	allocatable, err := ed.plugin.host.GetNodeAllocatable()
```

Comment:
we would tend to have limits always specified on the pod where isolation was a concern.  Wouldn't eviction help here in pathological case?  We fail a pod in that case, we would want to ensure we use pod level cgroup rather than sum container cgroups soon.  A failed pod will cause the volume to be cleaned up.